balances-adjustment/src/main/java/com/powsybl/balances_adjustment/balance_computation/BalanceComputationImpl.java

```diff
@@ -64,20 +64,20 @@ public BalanceComputationImpl(List<BalanceComputationArea> areas, ComputationMan
      */
     @Override
     public CompletableFuture<BalanceComputationResult> run(Network network, String workingStateId, BalanceComputationParameters parameters) {
-        return this.run(network, workingStateId, parameters, Reporter.NO_OP);
+        return this.run(network, workingStateId, parameters, ReportNode.NO_OP);
     }
 
     /**
      * Run balances adjustment computation in several iterations
      */
     @Override
-    public CompletableFuture<BalanceComputationResult> run(Network network, String workingStateId, BalanceComputationParameters parameters, Reporter reporter) {
+    public CompletableFuture<BalanceComputationResult> run(Network network, String workingStateId, BalanceComputationParameters parameters, ReportNode reportNode) {
         Objects.requireNonNull(network);
         Objects.requireNonNull(workingStateId);
         Objects.requireNonNull(parameters);
-        Objects.requireNonNull(reporter);
+        Objects.requireNonNull(reportNode);
 
-        BalanceComputationRunningContext context = new BalanceComputationRunningContext(areas, network, parameters, reporter);
+        BalanceComputationRunningContext context = new BalanceComputationRunningContext(areas, network, parameters, reportNode);
```

Comment:
Just wondering why we give two reportNodes to the context even though it's useless :raised_eyebrow: Not the purpose of this PR anyway!

Reply:
Agree, it is not clear that the two report nodes are needed. I just kept the same behaviours, just upgrading to the new report API.

Reply:
@zamarrenolm you may clarify via comments that (please adjust with new reporting api terminology, unsure of correct new terms ...):
- reportNode is the root reporter
- iterationReportNode is subreporter for the current balancing iteration

Reply:
Updated with comments. Also used the method to obtain the root report node from the context, to make explicit that the balance adjustment different steps have it available through the context as eventually the complexity of the current calculation could be split in many lower level methods.

Reply:
Thanks Luma for clarifying with comments. Nonetheless I think it would be clearer to remove the two `ReportNode`s from the context, as hiding them in this object does not help understanding the reports split... but again it's not the purpose of current PR!

Reply:
@flo-dup I don't understand previous comments '... it's useless ...', 'it would be clearer to remove the two ReportNodes', '... hiding ...'

The two report nodes are different, they are both useful, and should not be removed.
They are not hidden in the context, it is the opposite, they are made available via the context, so they can actually be used.